### PR TITLE
Fixed SoftmaxCrossEntropyLoss with spare labels

### PR DESCRIPTION
**Pull request by ifeherva**
SoftmaxCrossEntropyLoss in gluoncv currently fails with sparse labels. This PR fixes is the same way mxnet gluon does it.

### CODE REVIEW

**Comment by mli** [comment]
Job PR-809-1 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-809/1/index.html 
Code coverage of this PR:  vs. Master:

File: gluoncv/loss.py [hunk 1]
```diff
@@ -278,7 +278,11 @@ def hybrid_forward(self, F, pred, label):
             pred, label.astype(pred.dtype), ignore_label=self._ignore_label,
             multi_output=self._sparse_label,
             use_ignore=True, normalization='valid' if self._size_average else 'null')
-        loss = -F.pick(F.log(softmaxout), label, axis=1, keepdims=True)
+        if self._sparse_label:
+            loss = -F.pick(F.log(softmaxout), label, axis=1, keepdims=True)
+        else:
+            label = _reshape_like(F, label, pred)
+            loss = -F.sum(pred * label, axis=-1, keepdims=True)
```
[the remark this answers is not in the document]
Comment (ifeherva) [reply gluoncv/loss.py]: Done. Thanks for catching it.

[the remark this answers is not in the document]
Comment (zhanghang1989) [reply gluoncv/loss.py]: Thanks! Let's wait for the CI to finish again.

**Comment by mli** [comment]
Job PR-809-2 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-809/2/index.html 
Code coverage of this PR:  vs. Master:

**Comment by hetong007** [comment]
@zhanghang1989 could you please update?

**Comment by Maktu6** [comment]
https://github.com/dmlc/gluon-cv/blob/276ffba742d4cfe51336a76b702647c52ebb6ee0/gluoncv/loss.py#L285

I wonder `axis` should be 1. 
Additionally, why not use `F.log_softmax(pred, self._axis)` directly which is used in [here](https://mxnet.incubator.apache.org/_modules/mxnet/gluon/loss.html#SoftmaxCrossEntropyLoss)?
